purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -60,38 +60,34 @@ def _compute_invoice(self):
         for order in self:
             order.invoice_count -= order.invoice_refund_count
 
-    @api.multi
     def action_view_invoice_refund(self):
         """This function returns an action that display existing vendor refund
         bills of given purchase order id.
         When only one found, show the vendor bill immediately.
         """
-        action = self.env.ref("account.action_vendor_bill_template")
+        action = self.env.ref("account.action_move_in_refund_type")
         result = action.read()[0]
         create_refund = self.env.context.get("create_refund", False)
         refunds = self.invoice_ids.filtered(lambda x: x.type == "in_refund")
         # override the context to get rid of the default filtering
         result["context"] = {
-            "type": "in_refund",
-            "default_purchase_id": self.id,
-            "default_currency_id": self.currency_id.id,
+            "default_type": "in_refund",
             "default_company_id": self.company_id.id,
-            "company_id": self.company_id.id,
+            "default_purchase_id": self.id,
         }
         # choose the view_mode accordingly
         if len(refunds) > 1 and not create_refund:
             result["domain"] = "[('id', 'in', " + str(refunds.ids) + ")]"
         else:
             res = self.env.ref("account.invoice_supplier_form", False)
             result["views"] = [(res and res.id or False, "form")]
-            # Do not set an invoice_id if we want to create a new refund.
+            # Do not set an move_id if we want to create a new refund.
             if not create_refund:
                 result["res_id"] = refunds.id or False
         result["context"]["default_origin"] = self.name
         result["context"]["default_reference"] = self.partner_ref
         return result
 
-    @api.multi
     def action_view_invoice(self):
         """Change super action for displaying only normal invoices."""
         result = super(PurchaseOrder, self).action_view_invoice()
```

Comment:
This should be like other `super()` on code. It is not mandatory, but I think that to maintain consistency, it is the most correct.
```suggestion
        result = super().action_view_invoice()
```